components/support/migration/src/main/java/mozilla/components/support/migration/FennecMigrator.kt

```diff
@@ -709,4 +737,35 @@ class FennecMigrator private constructor(
             }
         }
     }
+
+    @SuppressWarnings("TooGenericExceptionCaught")
+    private suspend fun migrateAddons(): Result<AddonMigrationResult> {
+        return try {
+            logger.debug("Migrating add-ons...")
+            val result = AddonMigration.migrate(engine!!)
+            if (result is Result.Failure<AddonMigrationResult>) {
+                val migrationFailureWrapper = result.throwables.first() as AddonMigrationException
+                return when (val failure = migrationFailureWrapper.failure) {
+                    is AddonMigrationResult.Failure.FailedToQueryInstalledAddons -> {
+                        logger.error("Failed to query installed add-ons: $failure")
+                        crashReporter.submitCaughtException(migrationFailureWrapper)
+                        result
+                    }
+                    is AddonMigrationResult.Failure.FailedToMigrateAddons -> {
+                        logger.error("Failed to migrate add-ons: $failure")
+                        crashReporter.submitCaughtException(migrationFailureWrapper)
```

Comment:
I don't think we'll actually see the collected underlying exceptions in Sentry, right? In a similar situation in the logins exception, I collapse collected exceptions into a set and submit them individually - e.g. see `collectExceptionsIntoASet`

Reply:
Ah, yes, I remember this. Case here is slightly different as I am mapping the exceptions to their addon, but I made part of your logic re-usable and applied it here as well so we now
- properly report the underlying exception
- and only do it once (and not spam Sentry)

Reply:
Sounds great!